src/main/java/tatracker/model/TaTracker.java

```diff
@@ -265,13 +258,34 @@ public void setGroup(Group target, Group editedGroup, Module targetModule) {
      * Replaces the contents of the group list with {@code groups}.
      * {@code groups} must not contain duplicate groups.
      */
-    public void setGroups(List<Group> groups) {
-        this.groups.setGroups(groups);
+    public void setCurrentlyShownGroups(List<Group> groups) {
+        System.out.println("inside ta tracker 2");
+        this.currentlyShownGroups.setGroups(groups);
+        System.out.println("inside ta tracker 3");
+    }
+
+    /**
+     * Replaces the contents of the group list with the groups at the given
+     * module index.
+     */
+    public void setCurrentlyShownGroups(int n) {
+        System.out.println("inside ta tracker 2");
+        setCurrentlyShownGroups(((modules.get(n)).getGroupList()));
+        System.out.println("inside ta tracker 3");
+    }
+
+    /**
+     * Updates the currently shown groups to be that of the currently shown module
+     * code.
+     */
+    public void updateCurrentlyShownGroups(String moduleCode) {
+        System.out.println("inside ta tracker");
+        setCurrentlyShownGroups((modules.getModule(moduleCode)).getGroupList());
```

Comment:
Nice method names! Very clear what they are used for!

Reply:
Thanks! :)